### DISCUSSION TITLE
Event Bots

### DISCUSSION THREAD

**Discussion by gajop** [tonarino/bombercrab-player#6] [Show and tell]
Here's the [Hidari-kun](https://github.com/gajop/bombercrab-player/tree/a60216531e7924ec367d026b32eee3a607a89d7e) character I made during the event.

I developed it in these steps:
1. Simple AI that does random actions. I gave up on that almost immediately since adding the rand crate [doesn't work](https://github.com/rust-random/rand/issues/1110) for wasm targets, and I didn't want to explore the [workarounds](https://github.com/rust-random/rand/issues/1110#issuecomment-981604119), at least not for something that was meant to be a temporary solution.
2. "Go left" [wall follower](https://en.wikipedia.org/wiki/Maze-solving_algorithm#Wall_follower) (and drop bombs when direction changed) AI. I knew that this approach would be limited, but it felt simple enough that I could get it done in under 3hours. Things finally started to work, and on some maps (🇫🇮 ;)) I started to get some points.
3. Pathfind to closest found hill if possible, and pathfind to closest unexplored position otherwise. The second part (unexplored position search) was probably the most important element and something I should've started from the get-go. To do any kind of pathfinding, I also had to first create a map of the world. 

So how well did it do? I think I managed to win a few games out of roughly 60 :) Getting some points at least was the original goal, so I'm calling it a success!
![PXL_20220812_133957399 MP](https://user-images.githubusercontent.com/64901/184466317-9f960094-cccf-4e96-a3e6-c5014eccd7e5.jpg)

PS: Thanks for organizing the event. Despite (or maybe due to?) the time constraint and inability to debug things, it was a lot of fun. It was my first time coming to a Rust-related meetup, hopefully there will be more such things in the future.

**Comment by strohel** [comment]
Thanks for sharing your code, @gajop!

Code for my `MatBomber` bot is in https://github.com/tonarino/bombercrab-player/pull/5.
